include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -424,7 +424,11 @@ struct __sycl_usm_alloc
     operator()(::std::size_t __elements) const
     {
         const auto& __queue = __exec.queue();
-        return (_T*)sycl::malloc(sizeof(_T) * __elements, __queue.get_device(), __queue.get_context(), __alloc_t);
+        auto __buf = static_cast<_T*>(
```

Comment:
What if we a little bit rewrite this?

```C++
    _T*
    operator()(::std::size_t __elements) const
    {
        if (auto __buf = static_cast<_T*>(
                sycl::malloc(sizeof(_T) * __elements, __exec.queue().get_device(), __queue.get_context(), __alloc_t)))
            return __buf;

        throw std::bad_alloc();
    }
```

Reply:
This is mostly a preference of style, isn't it? Or do you feel strongly about this needing to be changed post merging?

Reply:
Not really, I think: most preferable to handle under `if (true)`  most frequently case. That's what I mean.